pkg/controllers/csi/metadata/models.go

```diff
@@ -37,15 +37,15 @@ type OSMount struct {
 	VolumeMeta VolumeMeta
 	TimeStampedModel
 	TenantUUID    string `gorm:"primaryKey" json:"tenantUUID,omitempty"`
-	VolumeMetaID  string
+	VolumeMetaID  string `gorm:"not null"`
 	Location      string `gorm:"not null"`
 	MountAttempts int64  `gorm:"not null"`
 }
 
 // AppMount keeps track of our mounts to user applications, where we provide the codemodules.
 type AppMount struct {
 	VolumeMeta VolumeMeta
-	CodeModule CodeModule `gorm:"foreignKey:Version;references:CodeModuleVersion"`
+	CodeModule CodeModule `gorm:"foreignKey:CodeModuleVersion;references:Version"`
 	TimeStampedModel
 	VolumeMetaID      string
```

Comment:
also we are missing a primary key for AppMount- could we use `VolumeMetaID` for that? Would make OPS on an AppMount much easier imho...

Reply:
fixed.